src/Tools/Source/CompilerGeneratorTools/Source/CSharpSyntaxGenerator/Grammar/GrammarGenerator.cs

```diff
@@ -74,17 +68,30 @@ public static string Run(List<TreeType> types)
                 }
             }
 
+            // Add some rules not present in Syntax.xml.
+            AddLexicalRules(rules);
+
             // The grammar will bottom out with certain lexical productions. Create rules for these.
             var lexicalRules = rules.Values.SelectMany(ps => ps).SelectMany(p => p.ReferencedRules)
                 .Where(r => !rules.TryGetValue(r, out var productions) || productions.Count == 0).ToArray();
             foreach (var name in lexicalRules)
-                rules[name] = new List<Production> { new Production("/* see lexical specification */") };
+                rules[name] = [new("/* see lexical specification */")];
 
             var seen = new HashSet<string>();
 
             // Define a few major sections to help keep the grammar file naturally grouped.
-            var majorRules = ImmutableArray.Create(
-                "CompilationUnitSyntax", "MemberDeclarationSyntax", "TypeSyntax", "StatementSyntax", "ExpressionSyntax", "XmlNodeSyntax", "StructuredTriviaSyntax");
+            List<string> majorRules = [
+                "CompilationUnitSyntax",
+                "MemberDeclarationSyntax",
+                "TypeSyntax",
+                "StatementSyntax",
+                "ExpressionSyntax",
+                "XmlNodeSyntax",
+                "StructuredTriviaSyntax",
+                // Place all syntax tokens at the end to keep them out of the way.
+                "SyntaxToken",
+                .. rules["SyntaxToken"].SelectMany(r => r.ReferencedRules),
+                "Utf8Suffix"];
```

Comment:
this helps keep the grammar emitted in a sensible order (so you don't have tokens up in the member-decl section).